x/relationships/keeper/handler.go

```diff
@@ -31,6 +31,12 @@ func NewHandler(keeper Keeper) sdk.Handler {
 
 // handleMsgCreateRelationship handles the creation of a relationship
 func handleMsgCreateRelationship(ctx sdk.Context, keeper Keeper, msg types.MsgCreateRelationship) (*sdk.Result, error) {
+	// Check if the receiver has blocked the sender before
+	if keeper.IsUserBlocked(ctx, msg.Receiver, msg.Sender) {
+		return nil, sdkerrors.Wrap(sdkerrors.ErrInvalidRequest,
+			fmt.Sprintf("The user with address %s has been blocked from %s", msg.Sender, msg.Receiver))
```

Comment:
Since the user that's creating the relationship is the same that is blocked, we can use a more specific error message here:

```suggestion
			fmt.Sprintf("The user with address %s blocked you", msg.Receiver))
```